my_compassion/models/res_user.py

```diff
@@ -31,3 +34,153 @@ def reset_password(self, login):
+            if not user:
+                _logger.error(
+                    "Signup failed: No user found with login '%s' after signup.", login
+                )
+                return dbname, login, password  # Or handle as appropriate
+
+            # Ensure changes are flushed and committed before scheduling the job
+            self.flush()
+
+            # Queue the task to find or replace the partner
+            self.with_delay()._find_and_replace_partner_async(user.id, email)
+            _logger.info(
+                "Async task to find and replace partner for user %s queued.",
+                user.login,
+            )
+
+        return dbname, login, password
+
+    def action_reset_password(self):
+        """
+        Overriding to create and send a signup email via communication job.
+        create signup token for each user, and send their signup url by email
+        """
+
+        if self.env.context.get("install_mode", False):
+            return
+        if self.filtered(lambda user: not user.active):
+            raise UserError(_("You cannot perform this action on an archived user."))
+        # prepare reset password signup
+        create_mode = bool(self.env.context.get("create_user"))
+        expiration = False if create_mode else now(days=+1)
+
+        self.mapped("partner_id").signup_prepare(
+            signup_type="reset", expiration=expiration
+        )
+
+        # Retrieve the communication config
+        comm_config = self.env.ref(
+            "my_compassion.user_account_confirmation",
+            raise_if_not_found=False,
+        )
+        if not comm_config:
+            raise UserError(_("Communication configuration is missing."))
+
+        for user in self:
+            partner = user.partner_id
+            if not partner.email:
+                raise UserError(
+                    _(
+                        "Cannot send email: the partner %s has no email address.",
+                        partner.name,
+                    )
+                )
+
+            job = self.env["partner.communication.job"].create(
+                {
+                    "partner_id": user.partner_id.id,
+                    "config_id": comm_config.id,
+                    "auto_send": True,
+                }
+            )
+
+            # Attempt to send the communication
+            try:
+                job.send()
+                _logger.info(
+                    "Account confirmation email sent for user <%s> to <%s>",
+                    user.login,
+                    partner.email,
+                )
+            except Exception as e:
+                _logger.error(
+                    "Failed to send account confirmation email for user %s: %s",
+                    user.login,
+                    e,
+                )
+                raise UserError(
+                    _("An error occurred while sending the account confirmation email.")
+                ) from e
```

Comment:
We already [redefined this method here](https://github.com/CompassionCH/compassion-switzerland/blob/14.0/partner_communication_switzerland/models/res_users.py#L21), are you sure we need it here?